File: demo-iys-rest-scraping.java/src/main/java/demo/js/SkillsParser.java
package demo.js;

import demo.model.Source;
import jdk.nashorn.api.scripting.ScriptObjectMirror;

import javax.script.ScriptEngine;
import javax.script.ScriptEngineManager;
import javax.script.ScriptException;
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.util.List;

public class SkillsParser {

    private final ScriptObjectMirror scriptObjectMirror;

    public SkillsParser() throws FileNotFoundException, ScriptException {
        ScriptEngine scriptEngine = new ScriptEngineManager().getEngineByName("nashorn");
        scriptEngine.eval(new FileReader("src/main/resources/js/iysx.js"));

        scriptObjectMirror = (ScriptObjectMirror) scriptEngine.eval("IYSX");
    }

    public synchronized List<Source> parse(String skills) {
        String json = (String) scriptObjectMirror.callMember("decompress", skills);
        return SourceListSerializer.fromJson(json);
    }
}

File: demo-iys-rest-scraping.kotlin/src/main/kotlin/demo/js/SkillsParser.kt
package demo.js

import demo.model.Source
import jdk.nashorn.api.scripting.ScriptObjectMirror
import java.io.FileNotFoundException
import java.io.FileReader
import javax.script.ScriptEngineManager
import javax.script.ScriptException

class SkillsParser @Throws(FileNotFoundException::class, ScriptException::class)
constructor() {

    private val scriptObjectMirror: ScriptObjectMirror

    init {
        val scriptEngine = ScriptEngineManager().getEngineByName("nashorn")
        scriptEngine.eval(FileReader("src/main/resources/js/iysx.js"))

        scriptObjectMirror = scriptEngine.eval("IYSX") as ScriptObjectMirror
    }

    @Synchronized fun parse(skills: String): List<Source> {
        val json = scriptObjectMirror.callMember("decompress", skills) as String
        return SourceListSerializer.fromJson(json)
    }
}
